src/Simulation/CsharpGeneration/Resources/EntryPointDriver.cs

```diff
@@ -0,0 +1,355 @@
+            TryCreateOption(new[] { "--simulator", "-s" },
+                            () => @EntryPointAdapter.DefaultSimulator,
+                            "The name of the simulator to use.")
+                .Then(simulate.AddOption);
+            simulate.Handler = CommandHandler.Create<@EntryPointAdapter, string>(Simulate);
+
+            var root = new RootCommand(@EntryPointAdapter.Summary) { simulate };
+            foreach (var option in @EntryPointAdapter.Options) { root.AddGlobalOption(option); }
+
+            // Set the simulate command as the default.
+            foreach (var option in simulate.Options) { root.AddOption(option); }
+            root.Handler = simulate.Handler;
+
+            return await root.InvokeAsync(args);
+        }
+
+        /// <summary>
+        /// Simulates the entry point.
+        /// </summary>
+        /// <param name="entryPoint">The entry point adapter.</param>
+        /// <param name="simulator">The simulator to use.</param>
+        /// <returns>The exit code.</returns>
+        private static async Task<int> Simulate(@EntryPointAdapter entryPoint, string simulator)
+        {
+            switch (simulator)
+            {
+                case "ResourcesEstimator":
+                    var resourcesEstimator = new ResourcesEstimator();
+                    await entryPoint.Run(resourcesEstimator);
+                    Console.WriteLine(resourcesEstimator.ToTSV());
+                    break;
+                default:
+                    var (isCustom, createSimulator) = simulator switch
+                    {
+                        "QuantumSimulator" => (false, new Func<IOperationFactory>(() => new QuantumSimulator())),
+                        "ToffoliSimulator" => (false, new Func<IOperationFactory>(() => new ToffoliSimulator())),
+                        _ => (true, @EntryPointAdapter.CreateDefaultCustomSimulator)
+                    };
+                    if (isCustom && simulator != @EntryPointAdapter.DefaultSimulator)
+                    {
+                        DisplayCustomSimulatorError(simulator);
+                        return 1;
+                    }
+                    await DisplayEntryPointResult(entryPoint, createSimulator);
+                    break;
+            }
+            return 0;
+        }
+
+        /// <summary>
+        /// Runs the entry point on a simulator and displays its return value.
+        /// </summary>
+        /// <param name="entryPoint">The entry point adapter.</param>
+        /// <param name="createSimulator">A function that creates an instance of the simulator to use.</param>
+        private static async Task DisplayEntryPointResult(
+            @EntryPointAdapter entryPoint, Func<IOperationFactory> createSimulator)
+        {
+            var simulator = createSimulator();
+            try
+            {
+                var value = await entryPoint.Run(simulator);
+#pragma warning disable CS0184
```

Comment:
What warning is that?

Reply:
[CS0184](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0184):
> The expression can never be true because the variable you are testing is neither declared as type nor derived from type.

The return type of the entry point is known at compile time, so the C# compiler knows when `value is QVoid` is always false. But it's convenient to write this if statement so I don't need to generate different code depending on the return type.

Reply:
Thanks for the explanation!